eth/beacon/helpers.py

```diff
@@ -362,3 +340,104 @@ def get_block_committees_info(parent_block: 'BaseBeaconBlock',
         proposer_committee_size=proposer_committee_size,
         shards_committees=shards_committees,
     )
+
+
+def get_beacon_proposer_index(state: 'BeaconState',
+                              slot: int,
+                              epoch_length: int) -> int:
+    """
+    Return the beacon proposer index for the ``slot``.
+    """
+    shard_committees = get_shard_committees_at_slot(
+        state,
+        slot,
+        epoch_length,
+    )
+    try:
+        first_shard_committee = shard_committees[0]
+    except IndexError:
+        raise ValueError("shard_committees should not be empty.")
+
+    proposer_committee_size = len(first_shard_committee.committee)
+
+    if proposer_committee_size <= 0:
+        raise ValueError(
```

Comment:
Do you think `ValidationError` would be more appropriate for these?  My loose guideline has been:

- `ValidationError` for things that we *might* encounter in live network conditions (like someone sending us a transaction with an invalid signature).
- `ValueError` for things that indicate broken code (and that when they occur we address them directly rather than by `try/except` catching the `ValueError`)